src/lib/Components/Bidding/Screens/ConfirmBid.tsx

```diff
@@ -15,10 +16,61 @@ interface ConfirmBidProps extends ViewProperties {
     display: string
     cents: number
   }
+  relay?: RelayPaginationProp
+  navigator?: NavigatorIOS
 }
 
 export class ConfirmBid extends React.Component<ConfirmBidProps> {
+  placeBid() {
```

Comment:
We may want to replicate some of force's logic here, which is a little more complex than just placing a mutation given that we have to wait for gravity's bids to be processed before we can show you the results screen.

(for reference, the logic is [here](https://github.com/artsy/force/blob/master/src/desktop/apps/auction_support/client/bid_form.coffee#L44-L84))

- First, it does a [client-side check](https://github.com/artsy/force/blob/master/src/desktop/apps/auction_support/client/bid_form.coffee#L49) to make sure that your bid isn't below what we're minimally accepting (I think this case is fairly rare, but if you'd been hanging out on the bid picker screen while bidding advanced much farther, it's possible).
- Then it sends the request to gravity and [shows an error](https://github.com/artsy/force/blob/master/src/desktop/apps/auction_support/client/bid_form.coffee#L61) if there was an error placing your bid (I don't believe being "outbid" counts as an error here-- it's more if the sale is closed to bidding or there is an internal error)
- Then we poll for the bid to be processed. After that, we can [tell you that you have been outbid](https://github.com/artsy/force/blob/master/src/desktop/apps/auction_support/client/bid_form.coffee#L77)

In this scenario, the mutation will likely not return an error for an outbid case (since the bidder position _was_ successfully created), but when you fetch the bidder position during polling from the `me/bidder_positions` endpoint you should be able to retrieve the status.

Reply:
This is a good point; I think we should move as much of this logic into metaphysics as possible, though. If I understand the GraphQL approach to error-handling, the mutation should never produce an error, but always a result.